inbm/dispatcher-agent/tests/ai/snapshot_genai_unit_test.py

```diff
@@ -0,0 +1,2079 @@
+        # Create a mock for the Rebooter object
+        mock_rebooter = Mock()
+
+        # Create a DebianBasedSnapshot instance with the test snap_num
+        snapshot = DebianBasedSnapshot(Mock(), "update", Mock(), snap_num, False, False)
+
+        # Mock the _rollback_and_delete_snap method
+        snapshot._rollback_and_delete_snap = Mock()
+
+        # Mock the time.sleep function
+        with patch('time.sleep') as mock_sleep:
+            # Call the revert method with the test parameters
+            snapshot.revert(mock_rebooter, time_to_wait_before_reboot)
+
+        # Check if _rollback_and_delete_snap was called
+        if should_reboot:
+            snapshot._rollback_and_delete_snap.assert_called_once()
+        else:
+            snapshot._rollback_and_delete_snap.assert_not_called()
+
+        # Check if the rebooter's reboot method was called
+        if should_reboot:
+            mock_rebooter.reboot.assert_called_once()
+        else:
+            mock_rebooter.reboot.assert_not_called()
+
+        # Check if time.sleep was called with the correct parameter
+        if should_reboot:
+            mock_sleep.assert_called_once_with(time_to_wait_before_reboot)
+        else:
+            mock_sleep.assert_not_called()
+
+#DO NOT DELETE THIS LINE - TestDebianBasedSnapshotRollbackAndDeleteSnap
+'''
+ADD HUMAN FEEDBACK BELOW:
+
+'''
+class TestDebianBasedSnapshotRollbackAndDeleteSnap:
+    @pytest.mark.parametrize('snap_num, sota_rollback_return, delete_snapshot_return, expected_telemetry_calls', [
+        ("123", (0, None), (0, None), [("SOTA attempting rollback"), ("Rollback succeeded"), ("Snapshot cleanup succeeded")]),
+        ("123", (1, "Error message"), (0, None), [("SOTA attempting rollback"), ("SOTA rollback failed: Error message")]),
+        ("0", (0, None), (0, None), [("SOTA rollback skipped")]),
+        (None, (0, None), (0, None), [("SOTA rollback skipped")]),
+        ("abc", (0, None), (0, None), [("SOTA rollback skipped")]),
+        ("99999999999999999999", (0, None), (0, None), [("SOTA attempting rollback"), ("Rollback succeeded"), ("Snapshot cleanup succeeded")]),
+        ("-123", (0, None), (0, None), [("SOTA rollback skipped")]),
+    ])
+    def test_rollback_and_delete_snap(self, snap_num, sota_rollback_return, delete_snapshot_return, expected_telemetry_calls):
+        trtl = Mock()
+        trtl.sota_rollback.return_value = sota_rollback_return
+        trtl.delete_snapshot.return_value = delete_snapshot_return
+        dispatcher_broker = Mock()
+
+        snapshot = DebianBasedSnapshot(trtl, "update", dispatcher_broker, snap_num, False, False)
+
+        snapshot._rollback_and_delete_snap()
+
+        calls = [call[0][0] for call in dispatcher_broker.telemetry.call_args_list]
+        assert calls == expected_telemetry_calls
+
+#DO NOT DELETE THIS LINE - TestDebianBasedSnapshotTakeSnapshot
+'''
+ADD HUMAN FEEDBACK BELOW:
+
+'''
+class TestDebianBasedSnapshotTakeSnapshot:
```

Comment:
To go back to Pull Request Summary table, Click on the [link](https://github.com/intel/intel-inb-manageability/pull/489#)

<h4>For feedback purposes, please include both emoticon-based and qualitative feedback in your reply to this comment.</h4>

To go back to Pull Request Summary table, Click on the [link](https://github.com/intel/intel-inb-manageability/pull/489#)